### ISSUE TITLE
[Enhancement] 

### ISSUE THREAD

**Issue by neilyoung** [mucommander/mucommander#1303]
### Description

I would like to raise a request for something I'm badly missing since 1.3x. I always used MUC for navigating to a certain directory. Then I was hitting the terminal icon and since my configuration was like shown below...

<img width="607" alt="image" src="https://github.com/user-attachments/assets/19988880-2fe1-4cb5-8063-aef6d8107d84">

... my beloved terminal app opened and I was in the proper directory. Now this is still happening with this config, but in the background the commander switches to the terminal view.

Wouldn't it be possible to introduce a right mouse click action like "open terminal here", which executes the "open -a Terminal ." ?

Would be great

**Comment by pskowronek** [comment]
Custom command won't do the job for you? https://github.com/mucommander/mucommander/wiki/Custom-Commands (with $p)?

**Comment by neilyoung** [comment]
Well, it would and I have such a file at the right place I guess. I just seem to be to blonde to trigger it. Can't find it in the GUI

**Comment by neilyoung** [comment]
Is there a way to assign such a self defined command to a key, or menu or something?

**Comment by pskowronek** [comment]
> Is there a way to assign such a self defined command to a key, or menu or something?

"Well, the most common reason to do so is that custom commands will appear in your Open With... contextual menu. "

**Comment by neilyoung** [comment]
> > Is there a way to assign such a self defined command to a key, or menu or something?
> 
> "Well, the most common reason to do so is that custom commands will appear in your Open With... contextual menu. "
> 
> Can you check there?

You are correct, that works. But it is an abusing solution: You have to tag a file in order to get the "Open with" context menu.

OK, it works, but it is a crutch. I again would like to point out, that I think it is a regression. The configuration (image) suggests, that there is an alternative between the internal terminal (which is not the best solution either, it is full screen, always bright, no way to configure styles etc.) and yet another terminal solution. 

In fact since version 4.x this configuration makes no sense anymore, since in my case it does both: It opens the default terminal and the inbuilt in background. This was not the case with 3.x, but this is not available as ARM build for Mac and is very slow in execution. There is really a choice.

But you already refused to consider a return to the 3.x behaviour, so don't bother. https://github.com/mucommander/mucommander/issues/859#issuecomment-2323810258

**Comment by pskowronek** [comment]
> OK, it works, but it is a crutch. I again would like to point out, that I think it is a regression.

Regression in technical sense that a bug was introduced, or regression in sense of features available in muC?

> The configuration (image) suggests, that there is an alternative between the internal terminal (which is not the best solution either, it is full screen, always bright, no way to configure styles etc.) and yet another terminal solution.

2 things here:

1) You can style the terminal no problem - for example I have it black:

<img width="552" alt="image" src="https://github.com/user-attachments/assets/df48c402-0bb9-4d31-8de7-22cac959d6f2">

<img width="477" alt="image" src="https://github.com/user-attachments/assets/5a94c2ce-9155-487d-a9be-78357b50cd30">

further more, you can use slider to make it smaller vertically:

<img width="698" alt="image" src="https://github.com/user-attachments/assets/240d44df-6127-4721-8025-9510ad30a5b9">


2) If german translation is not accurate:

<img width="880" alt="image" src="https://github.com/user-attachments/assets/61e5e20b-bd07-4da1-9ad1-f6053a5a030f">

Please provide us a better german wording for: `Use default system shell` and `Use custom shell` - 'command prompt' in german is maybe not so accurate, but it doesn't suggest a different type of terminal neither.

> EDIT: Assigned it a hotkey. No file selection necessary.

so, is your problem addressed and now you have muC and Terminal your way?

**Comment by neilyoung** [comment]
Thanks for the detailed follow up and the information contained. Will check this out. 


> Regression in technical sense that a bug was introduced, or regression in sense of features available in muC?
> 

Well in versions before 1.4 there was just the system terminal opened as floating window above the muC. This was my understanding of correct behaviour for "Use custom command prompt".  Just now (more precisely since 1.4) I'm getting the system shell _AND_ the inbuilt terminal window with the custom setting and I'm just getting the inbuilt terminal if I choose "Use default system shell". In comparison to my understanding of a (for me) correct behaviour provided until 1.3  I would name this a regression in the sense of a bug introduced. If you of course have a completely different behaviour in mind as that what I have derived from experience, then this is another story :)

> 2 things here:
> 
> 1. You can style the terminal no problem - for example I have it black:

Interesting. Where is this "custom theme editor"? In the settings?

> 2. If german translation is not accurate:
> In English: `Use default system shell` and `Use custom shell` respectively.

It was never fully clear to me what the difference was, except that I saw what "my" difference was: I was getting a system shell with the "Use custom shell" settings. Only. I was fine with this. That way I can't really provide a better translation (even if I would like to help here), if I don't really know what is meant to be the behaviour to be expected. BTW: The german translation throughout muC is superior (as the whole app is, to not be misunderstood, I love it and I literally know it since the glorious times of Norton Commander). 

> > EDIT: Assigned it a hotkey. No file selection necessary.
> 
> so, is your problem addressed and now you have muC and Terminal your way?

Yes, my problem is addressed. I'm opening a system shell window with shift+command+T now and it is all as it was before.

**Comment by pskowronek** [comment]
> > 3\. If german translation is not accurate:
> >     In English: `Use default system shell` and `Use custom shell` respectively.
> 
> It was never fully clear to me what the difference was, except that I saw what "my" difference was: I was getting a system shell with the "Use custom shell" settings. Only. I was fine with this. That way I can't really provide a better translation (even if I would like to help here), if I don't really know what is meant to be the behaviour to be expected. BTW: The german translation throughout muC is superior (as the whole app is, to not be misunderstood, I love it and I literally know it since the glorious times of Norton Commander).

So if you now know it should be and what `shell` is, are you able to provide us with a better wording for German?

> 
> > > EDIT: Assigned it a hotkey. No file selection necessary.
> > 
> > 
> > so, is your problem addressed and now you have muC and Terminal your way?
> 
> Yes, my problem is addressed. I'm opening a system shell window with shift+command+T now and it is all as it was before.

Good!

**Comment by neilyoung** [comment]
> So if you now know what it should be and what shell is, are you able to provide us with a better wording for German of these two fields?

Yes, describe to me, what you mean by that.

BTW: Where is this "custom theme editor"? I'm running 1.5.2 and I can't find it :)

**Comment by pskowronek** [comment]
Assuming that everybody knows what `shell` is (https://en.wikipedia.org/wiki/Unix_shell vs https://en.wikipedia.org/wiki/Terminal_emulator,  https://en.wikipedia.org/wiki/List_of_terminal_emulators )

`Use default system shell` - it uses a default `shell` defined in the operating system, `bash` or `zsh`, or bunch of others
`Use custom shell` - override a default `shell` to other one, for example `fish` (https://www.freecodecamp.org/news/linux-shells-explained/)

Shell is not a Terminal emulator, it is the command interpreted (like CMD.exe under Windows)

**Comment by neilyoung** [comment]
With this clarification in mind the current german translation hits the nail.

I unfortunately still don't know, where I can find this theme editor...

**Comment by pskowronek** [comment]
> With this clarification in mind the current german translation hits the nail.

"Standard-System-Shell verwenden" and "Benutzerdefinierte Shell verwenden" - maybe more something like this and 'Shell' - is 'shell' used in German commonly?

**Comment by neilyoung** [comment]
I think devs understand "shell"

**Comment by ahadas** [comment]
sorry for jumping in late here but I saw this thread came to a resolution in regards to the main issue, and as for the translation, it seems we can't find a better phrasing so let's keep it. I'm closing this issue as not planned then.

btw, when following this discussion, I played with the custom shell and found that when changing the shell, the change didn't take effect immediately so [I've fixed that](https://github.com/mucommander/mucommander/pull/1305).
